src/Columns/ColumnVector.cpp

```diff
@@ -471,6 +477,79 @@ void ColumnVector<T>::insertRangeFrom(const IColumn & src, size_t start, size_t
     memcpy(data.data() + old_size, &src_vec.data[start], length * sizeof(data[0]));
 }
 
+
+DECLARE_AVX512VBMI2_SPECIFIC_CODE(
+template <size_t ELEMENT_SIZE>
+inline void compressStoreAVX512(const void *src, void *dst, const UInt64 mask)
+{
+    __m512i vsrc = _mm512_loadu_si512(src);
+    if (ELEMENT_SIZE == 1)
+        _mm512_mask_compressstoreu_epi8(dst, static_cast<__mmask64>(mask), vsrc);
+    else if (ELEMENT_SIZE == 2)
+        _mm512_mask_compressstoreu_epi16(dst, static_cast<__mmask32>(mask), vsrc);
+    else if (ELEMENT_SIZE == 4)
+        _mm512_mask_compressstoreu_epi32(dst, static_cast<__mmask16>(mask), vsrc);
+    else if (ELEMENT_SIZE == 8)
+        _mm512_mask_compressstoreu_epi64(dst, static_cast<__mmask8>(mask), vsrc);
+}
+
+template <typename T, typename Container, size_t SIMD_BYTES>
+inline void doFilterAligned(const UInt8 *& filt_pos, const UInt8 *& filt_end_aligned, const T *& data_pos, Container & res_data)
+{
+    static constexpr size_t VEC_LEN = 64;   /// AVX512 vector length - 64 bytes
```

Comment:
It's a bit confusing that the same constant (64) is referred to by SIMD_LENGTH and by VEC_LEN.

Reply:
For `SIMD_BYTES`, the `BYTES` is a little bit confusing, actually if refer to batch num of elements copied at a time, the bytes is not always 64 (maybe 64*2, 64*4, etc.). And the value is not always 64 (depends on mask bit size).

So I introduce `VEC_LEN` to make it as actual zmm vector length in bytes.